### ISSUE TITLE
DIfferences between generations parameters and evolve calls

### ISSUE THREAD

**Issue by brenoarosa** [esa/pagmo2#211]
I'm using xNES to optmize cec2014 functions and even through I set the memory parameter to True there are differences between setting generations parameter to a given number and calling evolve method the same amount of times.

I believe that the expected behavior when memory is set to True is to lead to the same results.
Am I missing something?

Working example:
```python
import pygmo as pg

prob = pg.problem(pg.cec2014(prob_id=2, dim=10))
algo = pg.algorithm(pg.xnes(gen=1000,  memory=True))
archi = pg.archipelago(1, algo=algo, prob=prob, pop_size=15)
archi.evolve(1)
archi.wait()
print(archi[0].get_population().champion_f[0])
# returns: 200.0000005063659

prob = pg.problem(pg.cec2014(prob_id=2, dim=10))
algo = pg.algorithm(pg.xnes(gen=1,  memory=True))
archi = pg.archipelago(1, algo=algo, prob=prob, pop_size=15)

for _ in range(1000):
    archi.evolve(1)
    archi.wait()

print(archi[0].get_population().champion_f[0])
# returns: 29444484947.441223
```

ps: the same behavior was observed with CMA-ES.

**Comment by bluescarni** [comment]
Pinging @darioizzo (although I believe he will be on vacation for a week or so still).

Is this with pagmo/pygmo 2.8, or are you using the latest git head?

**Comment by brenoarosa** [comment]
I tested with pygmo 2.8

**Comment by bluescarni** [comment]
I believe this was fixed in #203. I'll try later today and let you know.

**Comment by bluescarni** [comment]
@brenoarosa pagmo/pygmo 2.9 have been released, so I am going to close this report for the time being. Please re-open if still experience the same issue.
